monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,307 @@
+            )
+        if mask.shape[0] != 1:
+            raise ValueError(f"Suppose the mask only has one channel, but got {mask.shape[0]}")
+        if hover_map.shape[0] != 2:
+            raise ValueError(f"Suppose the hover map only has two channels, but got {hover_map.shape[0]}")
+
+        hover_h = hover_map[0:1, ...]
+        hover_v = hover_map[1:2, ...]
+
+        hover_h = (hover_h - min(hover_h)) / (max(hover_h) - min(hover_h))
+        hover_v = (hover_v - min(hover_v)) / (max(hover_v) - min(hover_v))
+        sobelh = self.sobel_gradient(hover_h)[0, ...]
+        sobelv = self.sobel_gradient(hover_v)[1, ...]
+        sobelh = 1 - (sobelh - min(sobelh)) / (max(sobelh) - min(sobelh))
+        sobelv = 1 - (sobelv - min(sobelv)) / (max(sobelv) - min(sobelv))
+
+        # combine the h & v values using max
+        overall = maximum(sobelh, sobelv)
+        overall = overall - (1 - mask)
+        overall[overall < 0] = 0
+
+        return convert_to_dst_type(overall, mask, dtype=np.float32)[0]
+
+
+class GenerateDistanceMap(Transform):
+    """
+    Generate distance map.
+    In general, the instance map is calculated from the distance to the background.
+    Here, we use 1 - "foreground probability map" to generate the distance map.
+    Nuclei values form mountains so inverse to get basins.
+
+    Args:
+        smooth_fn: execute smooth function on distance map. Defaults to None. You can specify
+            callable functions for smoothing.
+            For example, if you want apply gaussian smooth, you can specify `smooth_fn = GaussianSmooth()`
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(self, smooth_fn: Optional[Callable] = None) -> None:
+        self.smooth_fn = smooth_fn
+
+    def __call__(self, mask: NdarrayOrTensor, foreground_prob_map: NdarrayOrTensor) -> NdarrayOrTensor:  # type: ignore
+        """
+        Args:
+            mask: binarized segmentation result. Shape must be [1, H, W].
+            foreground_prob_map: foreground probability map. Shape must be [1, H, W].
+        """
+        if mask.shape[0] != 1 or mask.ndim != 3:
+            raise ValueError(f"Input mask should be with size of [1, H, W], but got {mask.shape}")
+        if foreground_prob_map.shape[0] != 1 or foreground_prob_map.ndim != 3:
+            raise ValueError(
+                f"Input foreground_prob_map should be with size of [1, H, W], but got {foreground_prob_map.shape}"
+            )
+
+        distance_map = (1.0 - foreground_prob_map) * mask
+
+        if callable(self.smooth_fn):
+            distance_map = self.smooth_fn(distance_map)
+
+        return convert_to_dst_type(-distance_map, mask, dtype=np.float32)[0]
+
+
+class GenerateMarkers(Transform):
+    """
+    Generate markers used in `watershed`. Generally, The maximum of this distance (i.e., the minimum of the opposite of
```

Comment:
```suggestion
    Generate markers to be used in `watershed`. Generally, The maximum of this distance (i.e., the minimum of the opposite of
```

Reply:
I am not sure what do you mean by "this distance"? can you elaborate?

Reply:
Hi @drbeh, sorry for my lack of expression. I have updated the docstring in this [commit](https://github.com/Project-MONAI/MONAI/pull/5173/commits/cf61aa7908f4fc5d06cb9c86725b4c51b5c57d91). Hope it can be more understandable.